docs/configuring-a-resource.md

```diff
@@ -22,126 +22,239 @@ cloud provider; sometimes it could simply be the name of resource
 (e.g. VPC id ). This is something specific to resource, and we need some input
 configuration for terrajet to appropriately generate a resource.
 
-Since Terraform already needs the same identifier to import a resource, most
+Since Terraform already needs a similar [identifier to import a resource], most
 helpful part of resource documentation is the [import section].
 
-This is [the struct that holds the External Name configuration]:
+Terrajet performs some back and forth conversions between Crossplane resource
+model and Terraform configuration. We need a custom, per resource configuration
+to adapt Crossplane `external name` from Terraform `id`.
+
+![external name configuration](images/terrajet-externalname.png)
```

Comment:
nitpick: my two cents, even though the information on this diagram is needed only for the third case, it makes the whole thing look a bit more complicated if you never need the case 3, which I doubt will happen frequently in non-big three providers. Moving to under case 3 or removing would be my suggestion.

Reply:
I remember that I had hard times to grasp why we need 3 different functions just to configure external name and visualizing it helps a lot not to got lost which is setting what. So, I see value with having this available somewhere.

However, I totally agree with your point that it is not a good initial experience. So, I will hide it as much as I can but make it still accessible from this document.